### PR TITLE
Tests for arrays of arrays

### PR DESCRIPTION
**Pull request by mcabbott**
This adds some tests on arrays of arrays. There are two possible perspectives here as to what's represented: 
- a block matrix, in which case corresponding vectors will contain vectors, or 
- a representation of some possibly noncommutative numbers as matrices, in which case vectors will also contain matrices.

This tries to test both of them.

There appear to be few such tests in LinearAlgebra itself, @mschauer points out these ones:

https://github.com/JuliaLang/julia/blob/871ec3f60d15ebcddab65a1dbb7bbcdb0ea98a7c/stdlib/LinearAlgebra/test/cholesky.jl#L280

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -65,3 +65,113 @@ end
...
+        V = [rand(Int8,2).+7im for i in 1:3]
+        W = [rand(Int8,2).+11im for j in 1:4]
+
+        # ordinary LinearAlgebra
+        @test M * N' == (N * M')'
+
+        @test M * W == (W' * M')' # recursive adjoint is important here!
+        @test V' * M == (M' * V)'
+
+        @test V' * V == dot(V,V) # here the behaviour of dot is not wrong
```
Comment (mschauer) [inline test/runtests.jl:164]: This is a „choose one“ situation and making dot useful for block matrices seems to be more useful in practice.

Comment (mcabbott) [reply test/runtests.jl:164]: I think this was the argument made, but I'm not convinced it's true. Block matrices work with block vectors, like this `V`, for which `V' * V` already does the right thing. A non-recursive `dot` which takes the adjoint of each element would also do the right thing here.

**Comment by timholy** [comment]
I don't have any objections to these tests. Anyone else? (@mschauer?)

**Comment by timholy** [comment]
I'd say merge this.

**Pull request merged**